prover/circuits/srs_store.go

```diff
@@ -122,33 +160,27 @@ func (store *SRSStore) GetSRS(ctx context.Context, ccs constraint.ConstraintSyst
 		return nil, nil, fmt.Errorf("could not find canonical SRS for curve %s and size %d", curveID, sizeCanonical)
 	}
 
-	// find the lagrange srs
-	var lagrangeSRS kzg.SRS
-	for _, entry := range store.entries[curveID] {
-		if !entry.isCanonical && entry.size == sizeLagrange {
-			lagrangeSRS = kzg.NewSRS(curveID)
-			data, err := os.ReadFile(entry.path)
-			if err != nil {
-				return nil, nil, err
-			}
-			if err := lagrangeSRS.ReadDump(bytes.NewReader(data)); err != nil {
-				return nil, nil, err
-			}
-			break
-		}
-	}
-
-	if lagrangeSRS == nil {
+	// wait for lagrange SRS to be loaded
+	if loadLagrangeErr := <-loadLagrangeErr; lagrangeSRS == nil || loadLagrangeErr != nil {
 		// we can compute it from the canonical one.
 		if sizeCanonical < sizeLagrange {
 			panic("canonical SRS is smaller than lagrange SRS")
 		}
 		logrus.Debugf("computing lagrange SRS from canonical SRS %d -> %d\n", sizeCanonical, sizeLagrange)
 		var err error
-		lagrangeSRS, err = toLagrange(canonicalSRS, sizeLagrange)
-		if err != nil {
+		if lagrangeSRS, err = toLagrange(canonicalSRS, sizeLagrange); err != nil {
 			return nil, nil, err
 		}
+		// save the lagrange SRS to disk
```

Comment:
In prod, we don't touch the SRS. Do we? Or if we do, we only read so I'd assume this is fine

Reply:
The only downside I see for option 2 is that we would have to add a few more files to `prover/prover-assets` for our tests to pass and I know we don't like committing too many files into the repo.

Reply:
Ah but this is on prover runtime? Then no, we should not write this to disk. This makes the whole system more complicated than needed especially if it happens on the distributed file system in prod. We don't want to deal with the ramification of that.

The cleaner approach would be;

a. Try loading it
b. If it does not, try recomputing it locally but without writing it on disk
c. If not possible, throw a human-readable error explaining what is going on

The persons that are going to see issues related to finding the SRS are the SRE so better make clean error message as it saves time in case of failed deployment.

Reply:
Something you could do is to enable this writing-to-disk behaviour thanks to a global variable.